### PR TITLE
Add Steam controller userspace driver

### PR DESCRIPTION
**Pull request by rnhmjoj**
###### Motivation for this change
Purely selfish reason: I'm getting one so this needs to be packaged.
I'll soon add a module as well.

###### Things done
- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test (none)
- [x] Tested compilation of all pkgs that depend on this change (none)
- [x] Tested execution of all binary files
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/python-modules/libusb1/default.nix [hunk 1]
```diff
@@ -0,0 +1,27 @@
+{ stdenv, lib, buildPythonPackage, fetchPypi, libusb1 }:
+
+buildPythonPackage rec {
+  name = "${pname}-${version}";
```
Comment (FRidh) [inline pkgs/development/python-modules/libusb1/default.nix]: name is not needed for `buildPython*`

File: pkgs/development/python-modules/libusb1/default.nix [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  version = "1.6.4";
+  pname = "libusb1";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "03b7xrz8vqg8w0za5r503jhcmbd1ls5610jcja1rqz833nf0v4wc";
+  };
+
+  patchPhase = lib.optionalString stdenv.isLinux ''
```
Comment (FRidh) [inline pkgs/development/python-modules/libusb1/default.nix]: `postPatch`

File: pkgs/development/python-modules/libusb1/default.nix [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    substituteInPlace usb1/libusb1.py --replace \
+      "ctypes.util.find_library(base_name)" "'${libusb1}/lib/libusb-1.0.so'"
```
Comment (FRidh) [inline pkgs/development/python-modules/libusb1/default.nix]: see `${stdenv.hostPlatform.extensions.sharedLibrary}`

File: pkgs/development/python-modules/libusb1/default.nix [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  '';
+
+  buildInputs = [ libusb1 ];
+
+  meta = with stdenv.lib; {
+    homepage    = https://github.com/vpelletier/python-libusb1;
+    description = "Python ctype-based wrapper around libusb1";
+    license     = licenses.lgpl2Plus;
+    platform    = libusb1.meta.platforms;
```
Comment (FRidh) [inline pkgs/development/python-modules/libusb1/default.nix]: already set by `buildPython*`

**Review by FRidh — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/libusb1/default.nix (hunk 3 above)
Comment (rnhmjoj) [reply pkgs/development/python-modules/libusb1/default.nix]: That was fast! Thank you. I don't know how to use this. Any examples?

Comment (FRidh) [reply pkgs/development/python-modules/libusb1/default.nix]: `python-packages.nix` has some examples.

Comment (rnhmjoj) [reply pkgs/development/python-modules/libusb1/default.nix]: Ok, done.

**Comment by orivej** [comment]
I'm adding another driver in #33364 (originally based on this one) which additionally includes GUI configuration of profiles.

**Pull request merged**